posts/2022-09-01-java-19-22.0.0.10-beta.adoc

```diff
@@ -0,0 +1,291 @@
+
+1. link:https://jdk.java.net/19/[Download the early access release of Java 19].
+2. Modify your <<run, Maven coordinates>> for the 22.0.0.10-beta version of Open Liberty.
+3. Edit your Liberty server's link:{url-prefix}/docs/latest/reference/config/server-configuration-overview.html#server-env[server.env] file to point `JAVA_HOME` to your Java 19 installation.
+4. Start testing your apps!
+
+For more information on Java 19, visit the Java 19 link:https://jdk.java.net/19/release-notes[release notes page] and link:https://download.java.net/java/early_access/jdk19/docs/api/[API Javadoc page].
+
+As work is made towards full Java 19 support, there may be functionality that might not be 100% ready yet but is in progress to be complete soon!
+
+// https://github.com/OpenLiberty/open-liberty/issues/21847
+[#microprofile]
+== Create and manage traces, metrics, and logs using OpenTelemetry open standards (MicroProfile Telemetry 1.0)
+
+link:https://opentelemetry.io/[OpenTelemetry] is a set of APIs, SDKs, tooling and integrations that are designed for the creation and management of telemetry data such as traces, metrics, and logs. link:https://projects.eclipse.org/projects/technology.microprofile/releases/microprofile-telemetry-1.0/plan[MicroProfile Telemetry 1.0] adopts OpenTelemetry from CNCF to allow MicroProfile applications to benefit from OpenTelemetry and enable automatic tracing of Jakarta RESTful web services if configured.
+
+To use this feature update your `server.xml`: 
+[source, xml]
+----
+<featureManager>
+  <feature>mpTelemetry-1.0</feature>
+</featureManager>
+----
+
+You can view the telemetry data from your application in link:https://www.jaegertracing.io/docs/1.37/monitoring/#traces[Jaeger] and link:https://zipkin.io/pages/quickstart.html[Zipkin]. To export your traces to one of these exporters, configure the exporter with system properties, e.g. `-Dotel.traces.exporter=jaeger` or environment variables, e.g. `OTEL_TRACES_EXPORTER=jaeger`.  You can also configure the exporter endpoints using `OTEL_EXPORTER_JAEGER_ENDPOINT` and `OTEL_EXPORTER_ZIPKIN_ENDPOINT`.  To specify the name of the service, use `OTEL_SERVICE_NAME`.
+   
+For more information about MicroProfile Telemetry, see:
+
+* link:https://github.com/eclipse/microprofile-telemetry[Microprofile Telemetry]
+* link:https://github.com/open-telemetry/opentelemetry-specification/blob/v1.11.0/specification/trace/api.md[OpenTelemetry specification]
+* link:https://opentelemetry.io[opentelemetry.io]
+
+
+// https://github.com/OpenLiberty/open-liberty/issues/22092
+[#xml]
+== Repackaging of tools for XML binding (Jakarta XML Binding 4.0)
+
+Jakarta XML Binding 4.0 is an API for automating the mapping between XML documents and Java objects when writing XML documents in Java. The Jakarta XML Binding 4.0 feature provides the Jakarta XML Binding 4.0 tools for marshaling and unmarshaling XML to Java: `xjc` and `schemagen`, which are unchanged since Jakarta EE 9.

```

Comment:
```suggestion
Jakarta XML Binding 4.0 is an API for automating the mapping between XML documents and Java objects when writing XML documents in Java. By and large, the xmlBinding-4.0 feature is identical to xmlBinding-3.0, with a couple of exceptions. The 4.0 spec has removed the deprecated `jakarta.xml.bind.Validator API`. This means, if your application is using Validator to validate your JAX-B data types, you'll need to update to use the more common schema validation method. Here's an example of how to move from the Validator to schema validation: 
```
      // With Validator: 
      SomeJAXBDataOjbect dataObject = new SomeJAXBDataObject();

      JAXBContext context = JAXBContext.newInstance(new Class<>{]{ClassToBeBound.class, ClassToBeBound2.class, …});
      JAXBSource source = new JAXBSource(context, dataObject);

      Validator validator = schema.newValidator();
      validator.setErrorHandler(c);
      validator.validate(source);

      // With Schema Validation - Suggested alternative since 2.0. 

      JAXBContext context = JAXBContext.newInstance(new Class<>{]{ClassToBeBound.class, ClassToBeBound2.class, …});
      
      Unmarshaller junmarshaller = jaxbContext.createUnmarshaller();
       
      SchemaFactory schemaFactory = SchemaFactory.newInstance(XMLConstants.W3C_XML_SCHEMA_NS_URI);
      Schema dataObjectSchema = schemaFactory.newSchema(new File("dataObjectSchema.xml"));
      jaxbUnmarshaller.setSchema(dataObjectSchema);
       
      SomeJAXBDataObject dataObject = (SomeJAXBDataObject dataObject) umarshaller.unmarshal(new       File("dataObjectSchema.xml"));
```
In 4.0, you can now programmatically choose an alternative XML Binding Implementation. Provided you package the implementation with the application, then you simply need to set the name of the implementation in a property map, and pass the map along when instantiating a JAXBContext object. Here's an example doing just that with Eclipe's Moxy implementation:

```
       String JAXB_CONTEXT_FACTORY = "jakarta.xml.bind.JAXBContextFactory";
       String MOXY_JAXB_CONTEXT_FACTORY = "org.eclipse.persistence.jaxb.XMLBindingContextFactory";

       ClassLoader classLoader = some.package.with.jaxb.data.objects.ObjectFactory.class.getClassLoader();
       Map<String, String> properties = new HashMap<String, String>();
       properties.put(JAXB_CONTEXT_FACTORY, MOXY_JAXB_CONTEXT_FACTORY);
       JAXBContext context = JAXBContext.newInstance("some.package.with.jaxb.data.objects",
                                       classLoader, properties);
```